rpcserver.go

```diff
@@ -3107,10 +3137,48 @@ func (r *rpcServer) QueryAssetStats(ctx context.Context,
 		return nil, err
 	}
 
-	snapshots := assetStats.SyncStats
 	resp := &unirpc.UniverseAssetStats{
-		AssetStats: fn.Map(snapshots, marshalAssetSyncSnapshot),
+		AssetStats: make(
+			[]*unirpc.AssetStatsSnapshot, len(assetStats.SyncStats),
+		),
+	}
+	for idx, snapshot := range assetStats.SyncStats {
+		resp.AssetStats[idx] = marshalAssetSyncSnapshot(snapshot)
+		resp.AssetStats[idx].GenesisTimestamp = r.getBlockTimestamp(
+			ctx, snapshot.GenesisHeight,
+		)
 	}
 
 	return resp, nil
 }
+
+// getBlockTimestamp returns the timestamp of the block at the given height.
+func (r *rpcServer) getBlockTimestamp(ctx context.Context,
+	height uint32) int64 {
+
+	// Shortcut any lookup in case we don't have a valid height in the first
+	// place.
+	if height == 0 {
+		return 0
+	}
```

Comment:
This is to remain backward compatible, as we'll be running this patch in our testnet instance that will have proofs without a timestamp. So I think erroring out here is a bad idea, as a single invalid proof would break the whole stats RPC.